spec/membership_comparison_spec.rb

```diff
@@ -222,6 +225,54 @@
     specify { expect(comparison.problems['wds:1030-1DAA-4100']).to be_empty }
   end
 
+  context 'single existing P39, blank superclass position' do
+    let(:comparison) do
+      MembershipComparison.new(
+        existing:   {
+          'wds:1030-1DAA-4100' => { position: mp },
+        },
+        suggestion: suggestion_speaker
+      )
+    end
+
+    specify { expect(comparison.exact_matches).to be_empty }
+    specify { expect(comparison.partial_matches).to be_empty }
+    specify { expect(comparison.conflicts).to match_array('wds:1030-1DAA-4100') }
+    specify { expect(comparison.problems['wds:1030-1DAA-4100']).to match_array(['blank superclass position']) }
```

Comment:
Am I reading this correctly as saying that if someone has a blank P39 of "member of the UK Parliament" (for example), the tool will never add a "member of the 44th UK Parliament" statement? If so, then I don't think that's the desired behaviour. I would say it's OK to add a new statement in such a case.

Reply:
Yes, you're reading this correctly but we easily can change this. I think we would also need to define how partial matches are classified EG. if a statement for the superclass position has a start date which coincides with the suggestion would this then be a conflict? How about the other more generic party/district qualifiers; if we're changing the 'blank' case then my instinct would be say these aren't conflicts too?

Reply:
@tmtmtmtm I believe I've covered everything you've suggested in the latest changes. I code has also been refactored some hopefully is a bit more logical to understand. There is still the readability of the specs which we should open a ticket for. But the new specs cover the `bare`, `partial match`, `exact match` and `differing`  which should be all the cases we care about here?

Do you have any remaining questions/ thoughts?